packages/pdf-worker/src/strategies/ChatTranscript.ts

```diff
@@ -0,0 +1,88 @@
+import moment from 'moment-timezone';
+
+import type { Data } from '../types/Data';
+import type { IStrategy } from '../types/IStrategy';
+import exportChatTranscript from '../templates/ChatTranscript';
+import type { ChatTranscriptData, PDFMessage } from '../templates/ChatTranscript';
+
+export class ChatTranscript implements IStrategy {
+	private isNewDay(current: PDFMessage, previous: PDFMessage | undefined, timezone: string): boolean {
+		return !previous || !moment(current.ts).tz(timezone).isSame(previous.ts, 'day');
+	}
+
+	private parserMessages(messages: PDFMessage[], dateFormat: unknown, timeAndDateFormat: unknown, timezone: unknown): unknown[] {
+		return messages.map((message, index, arr) => {
+			const previousMessage = arr[index - 1];
+			const { ts, ...rest } = message;
+			const formattedTs = moment(ts)
+				.tz(timezone as string)
+				.format(String(timeAndDateFormat));
+			const isDivider = this.isNewDay(message, previousMessage, timezone as string);
+
+			if (isDivider) {
+				return {
+					...rest,
+					ts: formattedTs,
+					divider: moment(ts)
+						.tz(timezone as string)
+						.format(String(dateFormat)),
+				};
+			}
+
+			return {
+				...rest,
+				ts: formattedTs,
+			};
+		});
+	}
+
+	private getTranslations(translations: Record<string, unknown>[]): (key: string) => unknown {
+		return (key: string) => {
+			const translation = translations.find((t) => t.key === key);
+			if (!translation) {
+				throw new Error(`Translation not found for key: ${key}`);
+			}
+			return translation.value;
+		};
+	}
+
+	private isChatTranscriptData = (data: any): data is ChatTranscriptData => {
+		return (
+			'header' in data &&
+			'messages' in data &&
+			't' in data &&
+			'agent' in data.header &&
+			'visitor' in data.header &&
+			'siteName' in data.header &&
+			'date' in data.header &&
+			'time' in data.header
+		);
+	};
+
+	renderTemplate(data: Data): Promise<NodeJS.ReadableStream> {
+		if (!this.isChatTranscriptData(data)) {
+			throw new Error('Invalid data');
+		}
+		return exportChatTranscript(data);
+	}
+
+	parseTemplateData(data: Record<string, unknown | unknown[]>): Data {
```

Comment:
Talking with @KevLehman he informed me that the only `unknown` is this method, all other places data should be already known. Could you take another look at this file?

Reply:
Sure, but the thing is that the other functions rely on this data :/

Reply:
I did some changes to the file and tried to centralize the `unknown` data on this method only.